pallet/deposit/src/lib.rs

```diff
@@ -204,7 +201,7 @@ pub mod pallet {
 			let mut to_migrate = (0, Vec::new(), Vec::new());
 
 			// Take 0~10 deposits to migrate.
-			for d in deposits.by_ref().take(10) {
+			for d in deposits.by_ref().take(10).filter(|d| d.value != 0) {
```

Comment:
Will this filter make the deposit with value=0 being not take out and still leaving in the deposits?

This could still cause `<Deposits<T>>::iter().drain().next()` still be stuck at the user with the deposit which value is zero?

https://github.com/darwinia-network/darwinia/blob/7ed9ae01f10a7d0cd99af9b6ac73de0518856bf8/pallet/deposit/src/lib.rs#L140-L141

Reply:
https://github.com/darwinia-network/darwinia/pull/1636/files#diff-90778f3c64239af47f21d89b2cbcd4d68857be8120e317604dfc69456180538dR111-R125